### PR TITLE
py3(django): fix a few Django 1.9 deprecation warnings

### PR DESCRIPTION
**Pull request by joshuarli**
Note that these changes include all the Django 1.9 deprecation fixes except occurences like:

```
  /Users/joshua.li/dev/sentry/sentry/src/sentry/web/urls.py:58: RemovedInDjango19Warning: django.db.models.get_apps is deprecated.
    for app in get_apps():

  /Users/joshua.li/dev/sentry/sentry/src/sentry/web/urls.py:60: RemovedInDjango19Warning: django.db.models.get_models is deprecated.
    get_models(app)
```

These are in https://github.com/getsentry/sentry/pull/13796.

### CODE REVIEW

**Comment by joshuarli** [comment]
Some django-related dependencies need to be bumped to ensure compatibility with at least Django 1.9. Those are also being synced in https://github.com/getsentry/getsentry/pull/3227. So far, they don't need to be synced in sentry-plugins because it doesn't use/import them in plugin exclusive code.

**Comment by joshuarli** [comment]
Agh, wait. I'm seeing some more django 19 deprecated stuff that i'm not being warned about. Probably some uncovered code paths.

**Comment by joshuarli** [comment]
Actually, looks like for some reason I was doing it wrong. `PYTHONWARNINGS=always pytest --capture=no` will be verbose, perhaps too verbose (lots of deprecation warnings stemming from using 3rd party libs and even just the outdated django itself) but it looks like the only way to get pytest to spit out django deprecations during test running.

It's probably easier to fix these things in a branch where django is actually bumped to 1.9. Will get accurate feedback too.

File: src/social_auth/backends/__init__.py [hunk 1]
```diff
@@ -247,9 +247,7 @@ class BaseAuth(object):
 
     def __init__(self, request, redirect):
         self.request = request
-        # Use request because some auth providers use POST urls with needed
-        # GET parameters on it
-        self.data = request.REQUEST
+        self.data = request.GET.update(request.POST)
```
Comment (wedamija) [inline src/social_auth/backends/__init__.py]: Let's not modify this dict here. Could you make a copy first?

**Review by wedamija — APPROVED** [review]

File: src/social_auth/backends/__init__.py (hunk 1 above)
Comment (joshuarli) [reply src/social_auth/backends/__init__.py]: you right

**Pull request merged**